sdk/clientcore/core/src/main/java/io/clientcore/core/http/client/HttpClientProvider.java

```diff
@@ -32,13 +45,7 @@ public abstract class HttpClientProvider {
      * @return A shared instance of {@link HttpClient} that this {@link HttpClientProvider} is configured to
      * create.
      */
-    public final HttpClient getSharedInstance() {
-        if (sharedHttpClient == null) {
-            sharedHttpClient = getNewInstance();
-        }
-
-        return sharedHttpClient;
-    }
+    public abstract HttpClient getSharedInstance();
```

Comment:
Should we implement this method here with the default implementation? That way presumably `GlobalDefaultHttpClient` can be made private as well

Reply:
The reason `GlobalDefaultHttpClient` is package-private is so it can be used in `HttpClient.getSharedInstance()`:
```java
static HttpClient getSharedInstance() {
    return HttpClientProvider.getProviders().create(HttpClientProvider::getSharedInstance,
        HttpClientProvider.GlobalDefaultHttpClient.HTTP_CLIENT::getHttpClient, null);
}
```

As the `fallbackSupplier` in the `Providers.create()` method:

```java
public TInstance create(Function<TProvider, TInstance> createInstance, Supplier<TInstance> fallbackSupplier,
    Class<? extends TProvider> selectedImplementation)
```

It's possible I don't understand the `Providers` logic entirely, but it seems to me that if we want to keep the `HttpClient.getSharedInstance()` method functionality, `GlobalDefaultHttpClient` will still need to be exposed to `HttpClient` somehow to provide a `fallbackSupplier`.